### ISSUE TITLE
Incorrect author is listed on a change-less publish

### ISSUE THREAD

**Issue by NightJar** [silverstripe/silverstripe-versioned-snapshot-admin#51]
An publish action that makes no changes is recorded with a snapshot. The snapshot however lists the _previous_ author as the author for the newest snapshot.

This causes much confusion, particularly in cases concerning #50 - a common case as no page level content has changed, only related items reached via a cascading publish.

## Recreation

- A new project with versioned-snapshot-admin & 2 users (e.g. Default Admin & a Content Editor)
- `/admin/pages/edit/show/2` _click_ **Save & Publish**
- 👀  view history, see a single entry with no author at all. ❌ 
- Edit the page's content, save and publish again.
- 👀 view history, see a new entry listing the active user ✅ 
- _**Using the other user**_ click `Save & Publish` (without editing)
- 👀 view history, see a new entry listing the user that is **not** active (did not complete the latest publish). :x:

![image](https://user-images.githubusercontent.com/778003/115183158-5ebe5980-a12f-11eb-88b4-f32846b5d64d.png)
_User "Editor" (a "Content Author") publishes without changes_

![image](https://user-images.githubusercontent.com/778003/115183323-aa710300-a12f-11eb-98b0-44cffe226d36.png)
_User "Editor" (a "Content Author") publishes with changes_

![image](https://user-images.githubusercontent.com/778003/115183479-f2902580-a12f-11eb-8cdf-4e71b33e78ec.png)
_User "Default Admin" (an "Administrator") publishes without changes_

**Comment by unclecheese** [comment]
Should be resolved by: https://github.com/silverstripe/silverstripe-versioned-snapshot-admin/commit/a06a95fa5f935bf1e10d7316080778d28605845b

Author was being listed as the "origin version" author, but in changeless publish (or a publish of only "owned" content, e.g. blocks), there was no new version being created for the page, so the previous publisher was showing. Snapshot actions track their own authors and maintain that state independently.

**Comment by NightJar** [comment]
a06a95f indeed has resolved this issue for new entries.
However is _appears_ old entries have been lost to time by this change.
I have not downgraded this project to re-confirm there were values there to begin with yet.

![image](https://user-images.githubusercontent.com/778003/116874251-1fceee80-ac6d-11eb-8c6a-e8e7451872d6.png)

**Comment by NightJar** [comment]
Indeed, rolling back the testing updates has seen author information return.
Perhaps some kind of migration task or fallback lookup might be best
![image](https://user-images.githubusercontent.com/778003/116952642-4168be00-acdf-11eb-8d0f-cc68881a45f3.png)
